enterprise/server/raft/store/store.go

```diff
@@ -299,6 +299,37 @@ func (s *Store) RemoveRange(rd *rfpb.RangeDescriptor, r *replica.Replica) {
 	go s.releaseRangeLease(rd.GetRangeId())
 }
 
+func (s *Store) validatedRange(header *rfpb.Header) (*rfpb.RangeDescriptor, error) {
+	if header == nil {
+		return nil, status.FailedPreconditionError("Nil header not allowed")
+	}
+
+	s.rangeMu.RLock()
+	rd, rangeOK := s.openRanges[header.GetRangeId()]
+	s.rangeMu.RUnlock()
+	if !rangeOK {
+		return nil, status.OutOfRangeErrorf("%s: range %d", constants.RangeNotFoundMsg, header.GetRangeId())
+	}
+
+	if len(rd.GetReplicas()) == 0 {
+		return nil, status.OutOfRangeErrorf("%s: range had no replicas %d", constants.RangeNotFoundMsg, header.GetRangeId())
+	}
+
+	// Ensure the header generation matches what we have locally -- if not,
+	// force client to go back and re-pull the rangeDescriptor from the meta
+	// range.
+	if rd.GetGeneration() != header.GetGeneration() {
+		return nil, status.OutOfRangeErrorf("%s: generation: %d requested: %d", constants.RangeNotCurrentMsg, rd.GetGeneration(), header.GetGeneration())
+	}
+
+	return rd, nil
+}
+
+func (s *Store) rangeIsValid(header *rfpb.Header) error {
```

Comment:
nit: could you add comments explaining what Valid is (in contrast to Active) and when it should be used?